### PR TITLE
[UT] rm pytest.skip add within our team and use skiplist instead

### PR DESCRIPTION
**Pull request by AshburnLee**
Fix https://github.com/intel/intel-xpu-backend-for-triton/issues/1434

### CODE REVIEW

File: python/test/unit/language/test_core.py
Comment (whitneywhtsang) [inline python/test/unit/language/test_core.py:1]: Please do not remove skip from upstream.

Comment (AshburnLee) [reply python/test/unit/language/test_core.py:1]: Of course, I only remove those from our team.

Comment (whitneywhtsang) [reply python/test/unit/language/test_core.py:1]: e.g. the skips in test_dot under is_cuda is added from upstream, and we should not remove those.

Comment (AshburnLee) [reply python/test/unit/language/test_core.py:1]: I compared with the OpenAI Triton, and updated. thanks~

**Comment by pbchekin** [comment]
Have you compared the pass rate before and after? Ideally, we need the numbers to be the same or better (more passed tests with the same number of total tests).

**Comment by pbchekin** [comment]
```
llvm-target:
default: all: passed: 16680, failed: 0, skipped: 393, xfailed: 1224, total: 18297, fixme: 2, pass rate (w/o xfailed): 97.7%
```
```
lijunhui/rm_skip_rolling:
default: all: passed: 16675, failed: 0, skipped: 398, xfailed: 1224, total: 18297, fixme: 2, pass rate (w/o xfailed): 97.67%
```
The total number is the same, the number of passed tests is less than in the default branch. Do you have an explanation for this?

**Comment by AshburnLee** [comment]
- For `interpreter`:

```
lijunhui/rm_skip_rolling:
interpreter: passed: 6227, failed: 0, skipped: 28, xfailed: 687, total: 6942, fixme: 0, pass rate (w/o xfailed): 99.55%
```
```
llvm-target:
interpreter: passed: 6226, failed: 0, skipped: 29, xfailed: 687, total: 6942, fixme: 0, pass rate (w/o xfailed): 99.54%
```
Case: `test_precise_math[1-tl.math.sqrt_rn(x)-tl.math.sqrt(x.to(tl.float64)).to(tl.float32)]` passed in this branch while SKIPPED in `llvm-target`. There is no issue in `interpreter`

- For `language`:

```
lijunhui/rm_skip_rolling:
language: passed: 10356, failed: 0, skipped: 353, xfailed: 521, total: 11230, fixme: 0, pass rate (w/o xfailed): 96.7%
```
```
llvm-target:
language: passed: 10362, failed: 0, skipped: 347, xfailed: 521, total: 11230, fixme: 0, pass rate (w/o xfailed): 96.76%
```

In my branch, 6 cases passed ,yet I put them in the skip list:
```
test_core.py::test_reduce[4-32-1-xor_sum-bool-shape173-1-False]  
test_core.py::test_reduce[4-32-1-xor_sum-bool-shape175-1-False]
test_core.py::test_reduce[4-32-1-xor_sum-bool-shape177-1-False]
test_core.py::test_reduce[64-16-1-xor_sum-bool-shape173-1-False]
test_core.py::test_reduce[64-16-1-xor_sum-bool-shape175-1-False]
test_core.py::test_reduce[64-16-1-xor_sum-bool-shape177-1-False]
```

In fact, those 6 cases got fixed 2 days ago.

**Comment by AshburnLee** [comment]
Pass rate:
Rolling 97.7% -> 97.7%
LTS       96.94%->96.95%

**Review by whitneywhtsang — COMMENTED** [review]
@pbchekin is the same change needed for conda skiplist?

**Comment by pbchekin** [comment]
> @pbchekin is the same change needed for conda skiplist?

Good point.

**Comment by AshburnLee** [comment]
- Updated for `conda-build-test.yml`.
- No update for `conda-basekit-build-test.yml` as there's a error during this workflow, I've filed it in issue: https://github.com/intel/intel-xpu-backend-for-triton/issues/1562

**Pull request merged**